lib/web_ui/lib/src/engine/canvaskit/image_web_codecs.dart

```diff
@@ -0,0 +1,384 @@
+///
+/// See also:
+///
+///  * https://www.w3.org/TR/webcodecs/#imagedecoderesult-interface
+@JS()
+@anonymous
+class _DecodeResult {
+  external _VideoFrame get image;
+  external bool get complete;
+}
+
+/// Options passed to [_ImageDecoder.decode].
+///
+/// See also:
+///
+///  * https://www.w3.org/TR/webcodecs/#dictdef-imagedecodeoptions
+@JS()
+@anonymous
+class _DecodeOptions {
+  external factory _DecodeOptions({
+    required int frameIndex,
+  });
+}
+
+/// The only frame in a static image, or one of the frames in an animated one.
+///
+/// This class maps to the `VideoFrame` type provided by the browser.
+///
+/// See also:
+///
+///  * https://www.w3.org/TR/webcodecs/#videoframe-interface
+@JS()
+@anonymous
+class _VideoFrame {
+  external int allocationSize();
+  external JsPromise copyTo(Uint8List destination);
+  external String? get format;
+  external int get codedWidth;
+  external int get codedHeight;
+  external int get displayWidth;
+  external int get displayHeight;
+  external int? get duration;
+}
+
+/// Corresponds to the browser's `ImageTrackList` type.
+///
+/// See also:
+///
+///  * https://www.w3.org/TR/webcodecs/#imagetracklist-interface
+@JS()
+@anonymous
+class _ImageTrackList {
+  external JsPromise get ready;
+  external _ImageTrack? get selectedTrack;
+}
+
+/// Corresponds to the browser's `ImageTrack` type.
+///
+/// See also:
+///
+///  * https://www.w3.org/TR/webcodecs/#imagetrack
+@JS()
+@anonymous
+class _ImageTrack {
+  external int get repetitionCount;
+  external int get frameCount;
+}
+
+/// Represents an image file format, such as PNG or JPEG.
+class ImageFileFormat {
+  const ImageFileFormat(this.header, this.contentType);
+
+  /// First few bytes in the file that uniquely identify the image file format.
+  ///
+  /// Null elements are treated as wildcard values and are not checked. This is
+  /// used to detect formats whose header is split up into multiple disjoint
+  /// parts, such that the first part is not unique enough to identify the
+  /// format. For example, without this, WebP may be confused with .ani
+  /// (animated cursor), .cda, and other formats that start with "RIFF".
+  final List<int?> header;
+
+  /// The value that's passed as [_ImageDecoderOptions.type].
+  ///
+  /// The server typically also uses this value as the "Content-Type" header,
+  /// but servers are not required to correctly detect the type. This value
+  /// is also known as MIME type.
+  final String contentType;
+
+  /// All image file formats known to the Flutter Web engine.
+  ///
+  /// This list may need to be changed as browsers adopt new formats, and drop
+  /// support for obsolete ones.
+  ///
+  /// This list is checked linearly from top to bottom when detecting an image
+  /// type. It should therefore contain the most popular file formats at the
+  /// top, and less popular towards the bottom.
+  static const List<ImageFileFormat> values = <ImageFileFormat>[
+    // PNG
```

Comment:
AVIF is also supported by Chrome if you want.

Reply:
I do, but I don't know how to detect it :)

Reply:
It's slightly complicated if you want to do precise detection, but 'ftyp' in the first 16 bytes is good enough to detect an ISO-BMFF image type -- you'll fail later during configure if it's not avif and instead heic or a video.

Reply:
Fantastic! Thanks for the link. I'll add it.

Reply:
Done.